crates/polars-core/src/chunked_array/ops/aggregate/mod.rs

```diff
@@ -475,6 +475,49 @@ impl ChunkAggSeries for StringChunked {
     }
 }
 
+#[cfg(feature = "dtype-categorical")]
+impl CategoricalChunked {
+    fn min_categorical(&self) -> Option<&str> {
+        if self.is_empty() || self.null_count() == self.len() {
+            return None;
+        }
+        if self.uses_lexical_ordering() {
+            self.iter_str()
```

Comment:
We can include a fast path here. If all the categories in the rev_map (`can_fast_unique`) are used we only need to traverse the categories and not the whole chunkedarray

Reply:
Good point. Two follow-up:
1. I'm not sure I got the hint about `can_fast_unique`. Can you please share some more details?
2. Are you aware of any other case besides Enum where this fast path could be applied?
Thanks!

Reply:
There is a flag in the `bit_settings` field which tracks if all the categories are used. This allows the `unique` operator to be fast as it can immediately return the categories. 

You can check for this flag with `self.can_fast_unique()` which if true means all the categories are used. Which for the `min`/`max` case means you can apply the operator on the categories instead of the physicals. 

This fast path can be applied independent of dtype (so both `Enum` and `Categorical`

Reply:
Fast path added. Can you please have a look?